### PR TITLE
Gdr 1902

### PR DESCRIPTION
**Pull request by jjjermiah**
# Description Adding functionality to convert PSet to data.frame that can be used as input for the gDR pipeline. 
## What changed?
Related JIRA issue: 
  
## Why was it changed?
  
# Checklist for sustainable code base
- [ ] I added tests for any code changed/added
- [ ] I added documentation for any code changed/added
- [ ] I made sure naming of any new functions is self-explanatory and consistent

# Logistic checklist
- [ ] Package version bumped
- [ ] Changelog updated

# Screenshots (optional)

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -18,6 +18,7 @@ LazyLoad: yes
 Imports:
     assertthat,
     checkmate,
+    PharmacoGx,
```
Comment (marcin-kam) [inline DESCRIPTION:21]: We need to add new dependency to combo image. I've done that in https://code.roche.com/gdr/gdrshiny/-/merge_requests/3/diffs

File: inst/scripts/pset_to_gdr_TestScript.R [hunk 2]
```diff
@@ -0,0 +1,57 @@
+## 
+
+debugonce(gDRcore::create_SE)
```
Comment (marcin-kam) [inline inst/scripts/pset_to_gdr_TestScript.R:3]: Please remove

**Review by marcin-kam — APPROVED** [review]

File: R/pset_to_gdrDF.R [hunk 3]
```diff
@@ -0,0 +1,196 @@
+## 
+# collaboration with BHKLab
+# co-author: Jermiah Joseph
```
Comment (gladkia) [inline R/pset_to_gdrDF.R:3]: consider adding the info via the `@author` tag

File: R/pset_to_gdrDF.R [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+        PharmacoGx::downloadPSet(pset_name_param, saveDir = psetDir, timeout = timeout)
+      }
+    # Update PSet to new structure if not updated already.
+    PharmacoGx::updateObject(pset)
+}
+
+#' Get dose and viability readouts and melt into large data table
+#' @keywords internal
+#' @importFrom data.table `:=` setDF
+.extractDoseResponse <- function(pset) {
```
Comment (gladkia) [inline R/pset_to_gdrDF.R:109]: it would be great to stick to the 50 lines per function rule from BiocCheck

**Comment by gladkia** [comment]
Great job!

**Review by gladkia — APPROVED** [review]

File: R/pset_to_gdrDF.R (hunk 4 above)
Comment (bczech) [reply R/pset_to_gdrDF.R:101]: Done

File: inst/scripts/pset_to_gdr_TestScript.R (hunk 2 above)
Comment (bczech) [reply inst/scripts/pset_to_gdr_TestScript.R]: Done

File: R/pset_to_gdrDF.R [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+    checkmate::assert_flag(canonical)
+    checkmate::assert_numeric(timeout)
+
+    availPSets <- PharmacoGx::availablePSets(canonical = canonical)
+
+    pset_name_param <- if (pset_name %in% availPSets$"Dataset Name") {
+      availPSets[availPSets$"Dataset Name" == pset_name, "PSet Name"]
+      } else if (pset_name %in% availPSets$"PSet Name") {
+        pset_name    
+    } else {
```
[the remark this answers is not in the document]
Comment (bczech) [reply R/pset_to_gdrDF.R:90]: Done

**Pull request merged**